src/utilities/masked_coeffs.jl

```diff
@@ -0,0 +1,29 @@
+"""
+    MaskedCoeffs(coefficients)
+
+A mask for coefficients. Supports `getindex(::MaskedCoeffs, ijk...)`
+that forwards to coefficients, and `getindex(::Type{MaskedCoeffs}, ijk...)`
+that forwards (at compile time) to the mask, which behaves as a
+BitArray of the coefficients.
+"""
+struct MaskedCoeffs{S, m, C}
+    coeffs::C
+    function MaskedCoeffs(coeffs::C) where {C}
+        m = BitArray(iszero.(coeffs))
+        return new{size(m), Tuple(m), C}(coeffs)
+    end
+end
+
+# Forward array behavior:
+Base.@propagate_inbounds Base.getindex(mc::MaskedCoeffs, inds...) = @inbounds mc.coeffs[inds...]
+Base.length(mc::MaskedCoeffs) = length(mc.coeffs)
+import LinearAlgebra
+LinearAlgebra.diag(mc::MaskedCoeffs, args...) = LinearAlgebra.diag(mc.coeffs, args...)
+LinearAlgebra.adjoint(mc::MaskedCoeffs) = LinearAlgebra.adjoint(mc.coeffs)
+
+get_S(::MaskedCoeffs{S}) where {S} = S
+
+# Special behavior of MaskedCoeffs:
+Base.@propagate_inbounds Base.getindex(::Type{MaskedCoeffs{S, m, C}}, i::Int, j::Int) where {S, m, C} =
+    @inbounds m[i + S[1] * (j - 1)]
+Base.@propagate_inbounds Base.getindex(::Type{MaskedCoeffs{S, m, C}}, j::Int) where {S, m, C} = @inbounds m[j]
```

Comment:
Calling `getindex` on a `Type` is pretty confusing. We might want to make this a separate function, or just avoid it by storing non-zero indices in type space instead of the non-zero mask.

Reply:
Yeah, I'll define a custom method instead, it just looks very much like getindex.